src/SDK/Common/Configuration/Resolver/EnvironmentResolver.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Configuration\Resolver;
+
+use OpenTelemetry\SDK\Common\Configuration\Resolver;
+
+/**
+ * @internal
+ */
+class EnvironmentResolver extends Resolver
+{
+    public function hasVariable(string $variableName): bool
+    {
+        return getenv($variableName) !== false || isset($_SERVER[$variableName]);
+    }
+
+    /**
+     * @psalm-suppress InvalidReturnStatement
+     * @psalm-suppress InvalidReturnType
+     */
+    public function retrieveValue(string $variableName, ?string $default = ''): ?string
+    {
+        $value = getenv($variableName);
+        if ($value === false) {
+            $value = $_SERVER[$variableName] ?? $default;
```

Comment:
I think we should check `$_SERVER` first. For example vlucas/phpdotenv populates only `$_SERVER` and `$_ENV` by default, "[Using getenv() and putenv() is strongly discouraged due to the fact that these functions are not thread safe](https://github.com/vlucas/phpdotenv#putenv-and-getenv)".

Do we have to use [`local_only: true`](https://www.php.net/manual/en/function.getenv.php)?
> Warning If PHP is running in a SAPI such as Fast CGI, this function will always return the value of an environment variable set by the SAPI, even if [putenv()](https://www.php.net/manual/en/function.putenv.php) has been used to set a local environment variable of the same name. Use the local_only parameter to return the value of locally-set environment variables.

Reply:
I think if we want to remove getenv/putenv support it can be in a separate PR. I was worried about what that would mean for all the tests which rely on putenv, but I think that phpunit's `@backupGlobals` feature should do (and not having to support both $_ENV and getenv should simplify tests a little)